askMenuOption.cpp

```diff
@@ -1,23 +1,34 @@
-#pragma once
-#include <iostream>
-#include <map>
-
-#include "askMenuOption.h"
-
-void AskMenuOption::execute() {
-	std::cout << "Choose option: ";
-}
-
-PossibleState AskMenuOption::leave() {
-	int inputOption;
-	std::cin >> inputOption;
-
-	PossibleState nextState;
-	std::map<int, PossibleState> nextStateMap{
-		{1, PossibleState::addTask},
-		{2, PossibleState::showTasks},
-		{3, PossibleState::editTask}
-	};
-
-	return nextState = nextStateMap.at(inputOption);
+#include<iostream>
+#include<map>
+
+#include"askMenuOption.h"
+
+void AskMenuOption::nextStateSelection() {
+	int inputOption;
+	int* input_ptr;
+	input_ptr = &inputOption;
+	std::cout << "Choose option: ";
+	std::cin >> inputOption;
+	std::map<int, PossibleState> nextStateMap{
+		{1, PossibleState::addTask},
+		{2, PossibleState::showTasks},
+		{3, PossibleState::editTask},
+		{4, PossibleState::removeTask},
+		{5, PossibleState::terminate}
+	};
+	if(!nextStateMap.count(inputOption)){
+		std::cout << "Number out of bound!" << std::endl;
+		AskMenuOption::nextState = PossibleState::askMenuOption;
```

Comment:
nextState =
this->nextState = 

namespace raczej używamy do sięgania po pola klas bazowych